candidates/views/api.py

```diff
@@ -82,6 +84,82 @@ def get(self, request, *args, **kwargs):
         )
 
 
+class CandidatesAndElectionsForPostcodeView(APIView):
+    """
+    This re-produces a lot of UpcomingElectionsView, but the output
+    is different enough to justify it bing a different view
+    """
+
+    http_method_names = ['get']
+
+    def _error(self, error_msg):
+        return Response({'error': error_msg}, status=400,)
+
+    def get(self, request, *args, **kwargs):
+        postcode = request.GET.get('postcode', None)
+        coords = request.GET.get('coords', None)
+
+        # TODO: postcode may not make sense everywhere
+        errors = None
+        if not postcode and not coords:
+            return self._error('Postcode or Co-ordinates required')
+
+        try:
+            posts = fetch_posts_for_area(postcode=postcode, coords=coords)
+        except Exception as e:
+            return self._error(e.message)
+
+
+        election_kwargs = {
+            'current': True,
+        }
+        if 'all_elections' in request.GET:
+            del election_kwargs['current']
+        if 'date_gte' in request.GET:
+            if request.GET['date_gte'] == "today":
+                election_kwargs['election_date__gte'] = date.today()
+            else:
+                election_kwargs['election_date__gte'] = request.GET['date_gte']
+
+        results = []
+        for post in posts.all():
+            for election in post.extra.elections.filter(**election_kwargs):
+                candidates = []
+                for membership in post.memberships.filter(
+                        extra__election=election):
+                    candidates.append(
+                        serializers.PersonSerializer(
```

Comment:
I wonder if it would be worth creating a new serializer for this, if MinimalPersonSerializer doesn't provide enough? e.g. perhaps including the `versions` data structure for each candidate is more than is needed for this likely use case? (Having a less inclusive serializer could also help reduce the amount of queries needed for the page - at the moment requesting `/api/v0.9/candidates_for_postcode?postcode=BT17%200DF` causes 402 queries. Maybe "memberships" is another property that could be found by looking up the `url` field for the person, for example.)